app/controllers/key_pairs_controller.rb

```diff
@@ -2,14 +2,25 @@ class KeyPairsController < ApplicationController
   # GET /key_pairs
   # GET /key_pairs.json
   def index
-    @key_pairs = KeyPair.all
+    # Grab the own user's keys only
+    @key_pairs = KeyPair.where("user_id = ?", @current_user.id)
+
+    @key_strings = Array.new
+
+    @key_pairs.each do |keypair|
+      # Read the key
+      key = File.open(KEY_STORAGE + '/' + keypair.file_name)
+      @key_strings.push key.read
+    end
+
+    @key_pairs = @key_pairs.zip @key_strings
 
     respond_to do |format|
       format.html # index.html.erb
-      format.json { render json: @key_pairs }
+      format.json { render json => {:key_pairs => @key_pairs}}
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Space missing inside }.<br>Space inside { missing.<br>Space inside } missing.